File: AspNetMvcUnleashed/0672329980_source/Chapter21/CS/UnleashedBlog/UnleashedBlog/Attributes/AcceptAjaxAttribute.cs
using System;
using System.Reflection;
using System.Web.Mvc;

namespace UnleashedBlog.Attributes
{
    /// <summary>
    /// Allows an action to be called only within the context of an Ajax request. This
    /// code is from the Microsoft ASP.NET MVC futures.
    /// </summary>
    public sealed class AcceptAjaxAttribute : ActionMethodSelectorAttribute
    {
        public override bool IsValidForRequest(ControllerContext controllerContext, MethodInfo methodInfo)
        {
            if (controllerContext == null)
            {
                throw new ArgumentNullException("controllerContext");
            }
            return controllerContext.HttpContext.Request.IsAjaxRequest();
        }
    }
}

File: AspNetMvcUnleashed/0672329980_source/Chapter21/VB/UnleashedBlog/Attributes/AcceptAjaxAttribute.vb

Imports System.Reflection

''' <summary>
''' Allows an action to be called only within the context of an Ajax request. This
''' code is from the Microsoft ASP.NET MVC futures.
''' </summary>
Public NotInheritable Class AcceptAjaxAttribute
    Inherits ActionMethodSelectorAttribute

    Public Overrides Function IsValidForRequest(ByVal controllerContext As ControllerContext, ByVal methodInfo As MethodInfo) As Boolean
        If controllerContext Is Nothing Then
            Throw New ArgumentNullException("controllerContext")
        End If
        Return controllerContext.HttpContext.Request.IsAjaxRequest()
    End Function
End Class
